src/components/top_bar/top_bar.ts

```diff
@@ -188,6 +188,14 @@ export class TopBar extends Component<Props, SpreadsheetChildEnv> {
     this.closeMenus();
   }
 
+  onTopbarClick() {
+    if (this.env.model.getters.isReadonly()) {
```

Comment:
There's an unintended effect with this.
If you click anywhere on the top bar, let's say open a menu, the focus is no longer on the grid and you can move the selection with the keyboard anymore.
I think another way of solving this is to stop the click event on the top bar composant only with a `t-on-click.stop=""`

I think you can also try to write a test testing the focus is kept on the composer when clicking on it

Reply:
I have made the necessary changes, but there is a discrepancy in the test results. When clicking on the composer in the readonly mode, the developer tool shows it as the `activeElement`, but during test execution, the entire HTML body is obtained instead. Do you have any suggestions to explain this inconsistency?

![image](https://github.com/odoo/o-spreadsheet/assets/121081869/19867190-3376-46b0-a339-6e301bfd2feb)

![image](https://github.com/odoo/o-spreadsheet/assets/121081869/2b9f17db-9d00-4501-979f-5c41725a5a5c)

Reply:
Hmm I see. This doesn't surprise me.
In tests, it's not a full browser like Chrome or Firefox. It's a basic/partial implementation of web and DOM APIs for node js. It's https://github.com/jsdom/jsdom

Since it's only a simplified/partial implementation, some things does not properly work and doesn't reflect reality. For example, text selection is not implemented at all in jsdom. Focus is usually working, but maybe it's not working here.
Can you push the failing test and I'll have a look ?

Reply:
Okay, I have pushed the test, Thanks for your time!!!

Reply:
I checked but couldn't really understand what's going on. I don't want to spend too much time on this.
@rrahir wants to take a look.
If we can't find anything, we'll just remove the test (and pray it doesn't break in the future :crossed_fingers: )